fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -193,16 +198,20 @@ internal class HttpClient(
         )
     }
 
-    private fun dataStream(connection: HttpURLConnection): InputStream? {
+    private fun dataStream(connection: HttpURLConnection, hook: Hook?): InputStream? {
         return try {
             try {
-                BufferedInputStream(connection.inputStream)
+                val inputStream = hook?.interpretResponseStream(connection.inputStream) ?: connection.inputStream
+                BufferedInputStream(inputStream)
             } catch (_: IOException) {
                 // The InputStream SHOULD be closed, but just in case the backing implementation is faulty, this ensures
                 // the InputStream ís actually always closed.
                 try { connection.inputStream?.close() } catch (_: IOException) {}
 
-                connection.errorStream?.let { BufferedInputStream(it) }
+                connection.errorStream?.let {
+                    val errorInputStream = hook?.interpretResponseStream(it) ?: it
```

Comment:
Can you make a no-op implementation and remove nullability?

(It would pass the argument, so we don't need `hook?.interpretResponseStream(it) ?: it`, but just `hook.interpretResponseStream(it)`)